ibtk/include/ibtk/SecondaryHierarchy.h

```diff
@@ -0,0 +1,207 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2021 - 2021 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBTK_SecondaryHierarchy
+#define included_IBTK_SecondaryHierarchy
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibtk/config.h>
+
+#include "ibtk/SAMRAIDataCache.h"
+
+#include "BoxGeneratorStrategy.h"
+#include "GriddingAlgorithm.h"
+#include "IntVector.h"
+#include "LoadBalancer.h"
+#include "PatchHierarchy.h"
+#include "RefinePatchStrategy.h"
+#include "StandardTagAndInitStrategy.h"
+#include "tbox/Array.h"
+#include "tbox/Database.h"
+#include "tbox/Pointer.h"
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace IBTK
+{
+/**
+ * Class encapsulating the idea of a secondary hierarchy. In practice, this
+ * is an equivalent (same index space) hierarchy to a given PatchHierarchy,
```

Comment:
Is it equivalent in the sense that corresponding levels cover the same region?

Reply:
yup!

Reply:
My intent with that statement was to state that this should be the same patches in a different order.

Reply:
Is that what we actually want foe the typical use case?